pkg/controller/daemon/daemon_controller.go

```diff
@@ -1198,6 +1198,10 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *apps.DaemonSet, nodeL
 		return fmt.Errorf("error storing status for daemon set %#v: %v", ds, err)
 	}
 
+	// Resync the DaemonSet after MinReadySeconds as a last line of defense to guard against clock-skew.
+	if ds.Spec.MinReadySeconds > 0 && numberReady != numberAvailable {
+		dsc.enqueueDaemonSetAfter(ds, time.Duration(ds.Spec.MinReadySeconds)*time.Second)
```

Comment:
In replica sets, we ended up adding a second here because of an issue with time in golang. See:
https://github.com/kubernetes/kubernetes/blob/b3ad4cd6b99a413cc6da8ded20b535dd224007f5/pkg/controller/replicaset/replica_set.go#L351
https://github.com/kubernetes/kubernetes/issues/39785#issuecomment-279959133

Reply:
Interesting; this is already a thing, see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller.go#L561).

Reply:
https://github.com/kubernetes/kubernetes/blob/b3ad4cd6b99a413cc6da8ded20b535dd224007f5/pkg/controller/replicaset/replica_set.go#L621-L626

Reply:
> Interesting; this is already a thing, see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller.go#L561).

But when controller's clock time is slower than node, pod is still unavailable after minReadySeconds. For example:
pod A is ready at 12:00:00  and reenqueue at 12:00:31;
controller's time is 12:00:30 when reenqueue event happens, so controller judge A unavailable. 
So it's nessasary to add  the defense.

Reply:
> https://github.com/kubernetes/kubernetes/blob/b3ad4cd6b99a413cc6da8ded20b535dd224007f5/pkg/controller/replicaset/replica_set.go#L621-L626

daemon controller should give another chance to resync the DaemonSet in case of clock-skew, just like replicaset controller.

Reply:
/lgtm
/ok-to-test